### ISSUE TITLE
Python-Quart + Authentik?

### ISSUE THREAD

**Issue by ksaadDE** [goauthentik/authentik#4543]
Hello,

how to integrate Authentik using python3-quart ?

There's a "library" (plugin like) to use Keycloak, which was previously a more open library. But no clue if it's possible with Authentik.

I'm open to any idea. 

Regards

**Comment by BeryJu** [comment]
I haven't used quart myself, but using https://pypi.org/project/Quart-Session-OpenID/ should work just fine with authentik, creating an OAuth2 Provider and App with default settings, taking the client ID and secret from the provider, and the configuration URL from the provider page should be all that's required

**Comment by ksaadDE** [comment]
Hi, theoretically that's correct. 

> I haven't used quart myself, but using https://pypi.org/project/Quart-Session-OpenID/ should work just fine with authentik

The issue is, the package is **deprecrated**, as stated on PyPi. I don't want to use unmaintained / unsupported libs.

Quart-Session-OpenID is **Quart-Keycloak**
> Warning: this project is deprecated and replaced by https://pypi.org/project/Quart-Keycloak/

Please take a look at the Quart Lib and tell me what modifications I need to make, to get a "supported" plugin. 

I would like to provide a quart-authentik pack. I just have no clue about the underlying "mechanics" of authentik. That's why I need your help.

**Comment by ksaadDE** [comment]
> How do you intent to integrate it? 

I just want to integrate it into my project. So I can login across multiple micro-services using authentik. Basically custom SSO using the backend. 

However, I would like to know if the rest is working as well.

Not sure if authentik has a login-form too. It should, right? The docs stated that there's a proxy to all my other services possible. 

Using quart is less work, nice code. I love quart. Also, it improves the performance (asyncio).

**Comment by ksaadDE** [comment]
> Quart-Keycloak seems to use the same configuration as the old package and should work, with the configuration I mentioned above

Would need to try that. I had no time to test Authentik yet. I tried it first with Keycloak to see if the lib is even working.  I had hassle to get the lib running, to get the roles and groups into the access_token (mid-sized prob) and extracting them from the Token (smallest prob) using the lib.

So the lib indeed "works" with Keycloak. Despite being outdated on PyPi (Git Repo and PyPi diff) and it has a few not well-engineered parts, but it is still working. For example, I had to refactor multiple calls (three liners) of the same method, into one method and just calling that one new method "everywhere". 

I am testing all of that in a local docker-compose stack, so I had to disable TLS cert checks for local a dev env without exposing anyone to great risk. I did that by adding the ClientSession calls sslVerify=False to the mentioned refactored method. Finally, I had to edit the Dockerfile to move the modified file to overwrite the lib file in the container during build.

In case you have any chances to test the lib `quart-keycloak` with `Authentik` before I can,  a quick feedback about the results is highly appreciated.

Thanks for the help and I will keep you updated (vice versa?).
